worlds/cvcotm/__init__.py

```diff
@@ -0,0 +1,228 @@
+    get_start_inventory_data
+from .rom import RomData, patch_rom, get_base_rom_path, CVCotMProcedurePatch, CVCOTM_CT_US_HASH, CVCOTM_AC_US_HASH, \
+    CVCOTM_VC_US_HASH
+from .client import CastlevaniaCotMClient
+
+
+class CVCotMSettings(settings.Group):
+    class RomFile(settings.UserFilePath):
+        """File name of the Castlevania CotM US rom"""
+        copy_to = "Castlevania - Circle of the Moon (USA).gba"
+        description = "Castlevania CotM (US) ROM File"
+        md5s = [CVCOTM_CT_US_HASH, CVCOTM_AC_US_HASH, CVCOTM_VC_US_HASH]
+
+    rom_file: RomFile = RomFile(RomFile.copy_to)
+
+
+class CVCotMWeb(WebWorld):
+    theme = "stone"
+    options_presets = cvcotm_options_presets
+
+    tutorials = [Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up the Archipleago Castlevania: Circle of the Moon randomizer on your computer and "
+        "connecting it to a multiworld.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["Liquid Cat"]
+    )]
+
+    option_groups = cvcotm_option_groups
+
+
+class CVCotMWorld(World):
+    """
+    Castlevania: Circle of the Moon is a launch title for the Game Boy Advance and the first of three Castlevania games
+    released for the handheld in the "Metroidvania" format. As Nathan Graves, wielding the Hunter Whip and utilizing the
+    Dual Set-Up System for new possibilities, you must battle your way through Camilla's castle and rescue your master
+    from a demonic ritual to restore the Count's power...
+    """
+    game = "Castlevania - Circle of the Moon"
+    item_name_groups = {
+        "DSS": ACTION_CARDS.union(ATTRIBUTE_CARDS),
+        "Card": ACTION_CARDS.union(ATTRIBUTE_CARDS),
+        "Action": ACTION_CARDS,
+        "Action Card": ACTION_CARDS,
+        "Attribute": ATTRIBUTE_CARDS,
+        "Attribute Card": ATTRIBUTE_CARDS,
+        "Freeze": {iname.serpent, iname.cockatrice, iname.mercury, iname.mars},
+        "Freeze Action": {iname.mercury, iname.mars},
+        "Freeze Attribute": {iname.serpent, iname.cockatrice}
+    }
+    location_name_groups = get_location_name_groups()
+    options_dataclass = CVCotMOptions
+    options: CVCotMOptions
+    settings: typing.ClassVar[CVCotMSettings]
+    hint_blacklist = {lname.ba24}  # The Battle Arena reward, if it's put in, will always be a Last Key.
+
+    item_name_to_id = {name: cvcotm_item_info[name].code + BASE_ID for name in cvcotm_item_info
+                       if cvcotm_item_info[name].code is not None}
+    location_name_to_id = get_location_names_to_ids()
+
+    # Default values to possibly be updated in generate_early
+    total_last_keys: int = 0
+    required_last_keys: int = 0
+
+    auth: bytearray
+
+    web = CVCotMWeb()
+
+    def generate_early(self) -> None:
+        # Generate the player's unique authentication
+        self.auth = bytearray(self.multiworld.random.getrandbits(8) for _ in range(16))
```

Comment:
Should use world.random
```suggestion
        self.auth = bytearray(self.random.getrandbits(8) for _ in range(16))
```

Reply:
~Hey look, another thing I need to fix in CV64.~